[FIX] payment_stripe: make it work

nim-odoo opened the pull request:
The module `payment_stripe` was apparently not designed to work with
module `website_payment`, althought the payment method is available. For
example:
- Go to '/my/home', select "Pay Now" on an invoice
- Try to pay with Stripe ==> nothing happens

opw-693421

nim-odoo commented:
@jorenvo

nim-odoo commented:
@mart-e @jorenvo : I would appreciate your feedback on this

jorenvo commented on addons/payment_stripe/static/src/js/stripe.js:
```diff
@@ -41,27 +41,46 @@ odoo.define('payment_stripe.stripe', function(require) {
-                amount: $("input[name='amount']").val()*100
+        if ($('.o_website_payment')) {
+            ajax.jsonRpc('/website_payment/transaction', 'call', {
+                    reference: $("input[name='invoice_num']").val(),
```
I think through website_payment the acquirer form gets rendered through: https://github.com/odoo/odoo/blob/saas-11/addons/website_quote/controllers/main.py#L73. And so I think the reference will always be '/', leading to [a payment.transaction constraint](https://github.com/odoo/odoo/blob/saas-11/addons/payment/models/payment_acquirer.py#L415) triggering if you try to validate a second payment.

nim-odoo replied:
Hum your are pointing to `website_quote`, which is not necessarily installed. Moreover, the method clearly needs an `order_id`, while here we are paying an invoice. Have a look at https://github.com/odoo-dev/odoo/blob/saas-11-opw-693421-stripe-nim/addons/website_payment/static/src/js/payment_transaction.js#L21. The comment seems to match what we are doing:
`When clicking on payment button: create the tx using json then continue to the acquirer`

The reference is fine BTW, it gets the invoice number. If you try to pay the same invoice several time, it also adds a unique id at the end of the invoice number.

jorenvo commented on addons/payment_stripe/controllers/main.py:
```diff
@@ -51,4 +51,9 @@ def stripe_create_charge(self, **post):
         _logger.info('Stripe: entering form_feedback with post data %s', pprint.pformat(response))
         if response:
             request.env['payment.transaction'].sudo().form_feedback(response, 'stripe')
+        tx_transaction = request.env['payment.transaction'].browse(request.session.get('website_payment_tx_id', False))
```
This is kind of weird, I'm not sure there is a way around it but it essentially means we are creating two payment.transaction records for one single payment of eg. an online quotation?

nim-odoo replied:
Yeah, this is very weird, indeed. Maybe the trick is to NOT call `/website_payment/transaction` manually and let stripe create the transaction here. I just need it [here](https://github.com/odoo-dev/odoo/blob/saas-11-opw-693421-stripe-nim/addons/website_payment/controllers/main.py#L76), but it is only a matter of adding it in the context, so no biggie.

nim-odoo commented:
@jorenvo I've updated the PR after a discussion with @mart-e . There were actually 2 payment transactions created with Stripe, whatever the case (eCommerce, Online Quote or Website Payment). This was not necessary, so I removed that part, which leads to a cleaner code.

I tested the 3 use cases, and they seem to work fine now.
